### PR TITLE
Test/ Setup integration tests with ethers

### LINKED ISSUE #291: Integration tests setup.js: convert truffle -> ethers
Rewrite the `test/integration/setup.js` code to only use the ethers library and eliminate all truffle dependencies.

Currently the test creates all smart contracts and configures them using truffle objects, and at the end calls the `web3ToEthers` to convert all truffle objects to ethers.

The use of `web3ToEthers` should be completely removed.

All contracts and configs should be the exact same.

The interface exposed to the tests should be:  this.accounts, this.contracts so the this.withEthers nesting should be removed and we would no longer have `    const { ic, cover, stakingPool0, as } = this.withEthers.contracts;`

But simply: `    const { ic, cover, stakingPool0, as } = this.contracts;`

This applies currently to the `test/integration/IndividualClaims` directory.

For examples on how to handle everything with ethers please see the deploy script

https://github.com/NexusMutual/smart-contracts/blob/0e0ee5dd8479a360f985679010993825516cda11/scripts/deploy/deploy.js#L60

And unit tests:

https://github.com/NexusMutual/smart-contracts/blob/a6ef1ad7286305ddef2d6f4057a2c9dae8c612a1/test/unit/Cover/setup.js#L20

Keep namings in the new script the same as in the truffle code, unless there's good reasons to change them.

### PR DESCRIPTION
**Pull request by paterson1**
## Context

Closes #291

## Changes proposed in this pull request

Remove references to `withEthers`, `web3ToEthers`, `truffle`, `artifacts` and `web3` from integration test setup. `enrollMember` is failing and being caught by a catch clause to allow the tests to run.

## Test plan

Please describe the tests cases that you ran to verify your changes. Add further instructions on 
how to run them if needed (i.e. migration / deployment scripts, env vars, etc).

## Checklist

- [ ] Rebased the base branch
- [ ] Attached corresponding Github issue
- [ ] Prefixed the name with the type of change (i.e. feat, chore, test)
- [ ] Performed a self-review of my own code
- [ ] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [ ] Didn't generate new warnings
- [ ] Didn't generate failures on existing tests
- [ ] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: test/integration/setup.js [hunk 1]
```diff
@@ -455,10 +483,11 @@ async function setup() {
...
   this.contractType = contractType;
 
-  this.withEthers = web3ToEthers(this, signers);
-
-  await enrollMember(this.contracts, ethersAccounts.members, ethersAccounts.defaultSender);
-
+  try {
+    await enrollMember(this.contracts, ethersAccounts.members, ethersAccounts.defaultSender);
+  } catch (e) {
+    console.error(e);
```
Comment (danoctavian) [inline test/integration/setup.js:489]: need to throw again/quit the whole thing

**Comment by danoctavian** [comment]
Direction looks correct 👍 didn't get into details because it's still a draft.

But this is what we need, remove all truffle usage.

If all existing active integration tests pass then it's functionally correct as well.

File: test/integration/setup.js [hunk 2]
```diff
@@ -1,128 +1,153 @@
...
   // external
-  const WETH9 = artifacts.require('WETH9');
-  const CSMockSettlement = artifacts.require('CSMockSettlement');
-  const CSMockVaultRelayer = artifacts.require('CSMockVaultRelayer');
+  const WETH9 = await ethers.getContractFactory('WETH9');
+  const CSMockSettlement = await ethers.getContractFactory('CSMockSettlement');
+  const CSMockVaultRelayer = await ethers.getContractFactory('CSMockVaultRelayer');
+
+  const deployImmutable = async (contractFactory, constructorArgs = [], options = {}) => {
+    const { libraries } = options;
```
Comment (shark0der) [inline test/integration/setup.js:71]: The destructuring is not needed here. Rename `options` to `overrides` and pass it directly without the `if (libraries) ` condition.

File: test/integration/setup.js [hunk 3]
```diff
@@ -1,128 +1,153 @@
...
-  const QE = '0x51042c4d8936a7764d18370a6a0762b860bb8e07';
-  const INITIAL_SUPPLY = ether('15000000000');
+  const deployProxy = async (contract, deployParams = [], options = {}) => {
+    const { libraries } = options;
 
-  const deployProxy = async (contract, deployParams = []) => {
-    const implementation = await contract.new(...deployParams);
-    const proxy = await OwnedUpgradeabilityProxy.new(implementation.address);
-    return contract.at(proxy.address);
+    const contractFactory = await ethers.getContractFactory(contract, { libraries });
```
Comment (shark0der) [inline test/integration/setup.js:86]: Just like above, pass `overrides` directly as the last parameter.

**Pull request closed without merging**